app/services/foreman_ansible/inventory_creator.rb

```diff
@@ -31,7 +31,7 @@ def to_hash
     def hosts_vars
       hosts.reduce({}) do |hash, host|
         hash.update(
-          RemoteExecutionProvider.find_ip_or_hostname(host) => host_vars(host)
+          AnsibleProvider.find_ip_or_hostname(host) => host_vars(host)
```

Comment:
Same here: if we don't expect ip address to be here, let's call `find_fqdn` instead.